examples/adversarial_training.ipynb

```diff
@@ -0,0 +1,1052 @@
+        "  Madry, Aleksander, et al. \"Towards deep learning models resistant to\n",
+        "  adversarial attacks.\", https://arxiv.org/pdf/1706.06083.pdf"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "id": "xZ-lvFkUQ4N1"
+      },
+      "outputs": [],
+      "source": [
+        "%%capture\n",
+        "%pip install -U jax flax optax"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "id": "diO66z2ZQ4N3",
+        "outputId": "c1a68d1c-4101-481d-874b-441fd29bf611",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        }
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "TPU not found, continuing without it.\n"
+          ]
+        }
+      ],
+      "source": [
+        "import datetime\n",
+        "import collections\n",
+        "import operator\n",
+        "\n",
+        "# activate TPUs if available\n",
+        "try:\n",
+        "    import jax.tools.colab_tpu\n",
+        "    jax.tools.colab_tpu.setup_tpu()\n",
+        "except (KeyError, RuntimeError):\n",
+        "    print(\"TPU not found, continuing without it.\")\n",
+        "\n",
+        "import jax\n",
+        "from jax import numpy as jnp\n",
+        "from jax.tree_util import tree_map, tree_reduce\n",
+        "from flax import linen as nn\n",
+        "\n",
+        "from matplotlib import pyplot as plt\n",
+        "plt.rcParams.update({'font.size': 22})\n",
+        "\n",
+        "import optax\n",
+        "import tensorflow as tf\n",
+        "import tensorflow_datasets as tfds\n",
+        "# Hide any GPUs from TensorFlow. Otherwise TF might reserve memory and make\n",
+        "# it unavailable to JAX.\n",
+        "tf.config.experimental.set_visible_devices([], \"GPU\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "oWdSeE2_Q4N4"
+      },
+      "source": [
+        "Show on which platform JAX is running. The code below should take around 3 min to run on GPU but might take longer on CPUs."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 3,
+      "metadata": {
+        "id": "CyKEPpyoQ4N5",
+        "outputId": "34ca90b8-efe7-469c-bdc9-35853e6e8293",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        }
+      },
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "JAX running on GPU\n"
+          ]
+        }
+      ],
+      "source": [
+        "print(\"JAX running on\", jax.devices()[0].platform.upper())"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 4,
+      "metadata": {
+        "id": "HLuuYP6LQ4N6"
+      },
+      "outputs": [],
+      "source": [
+        "Flags = collections.namedtuple(\n",
```

Comment:
please use the new flags format we use for example in https://optax.readthedocs.io/en/latest/_collections/examples/cifar10_resnet.html (the cell that starts with @markdown). This format doesn't require to define the Flags namedtuple and instead stores everything as global variables